lib/logstash/inputs/log4j.rb

```diff
@@ -59,7 +57,25 @@ def register
     @logger.info("Log4j input")
   end # def register
 
-  public
+  def run(output_queue)
+    if server?
+      while !stop?
+        Thread.start(@server_socket.accept) do |s|
+          s.instance_eval { class << self; include ::LogStash::Util::SocketPeer end }
```

Comment:
On the other hand.. this mixin is only used for debugging and we could probably get rid of it and use a simple string interpolation for the debug.